libbeat/processors/add_cloud_metadata/provider_google_gce.go

```diff
@@ -63,11 +77,39 @@ var gceMetadataFetcher = provider{
 						"type": c.Str("machineType"),
 					},
 					"availability_zone": c.Str("zone"),
+					"orchestrator": s.Object{
+						"cluster": c.Dict(
+							"attributes",
+							s.Schema{
+								"name":       c.Str("cluster-name"),
+								"kubeconfig": c.Str("kubeconfig"),
+							}),
+					},
 				}.ApplyTo(out, instance)
 				trimLeadingPath("machine.type")
 				trimLeadingPath("availability_zone")
 			}
 
+			if kubeconfig, err := out.GetValue("orchestrator.cluster.kubeconfig"); err == nil {
+				kubeConfig, ok := kubeconfig.(string)
+				if !ok {
+					out.Delete("orchestrator.cluster.kubeconfig")
+				}
+				cc := &KubeConfig{}
+				err := yaml.Unmarshal([]byte(kubeConfig), cc)
+				if err != nil {
+					out.Delete("orchestrator.cluster.kubeconfig")
+				}
+				if len(cc.Clusters) > 0 {
+					if cc.Clusters[0].Cluster.Server != "" {
+						out.Delete("orchestrator.cluster.kubeconfig")
+						out.Put("orchestrator.cluster.url", cc.Clusters[0].Cluster.Server)
+					}
+				}
+			} else {
+				out.Delete("orchestrator")
```

Comment:
Maybe we can reorganize this processor so the fetchers return the cluster data under the `cluster` name and the orchestrator data under `orchestrator`, so we don't need to make so much mangling of this meta.
If we need to add another set of data apart of `orchestrator` this is going to be quite complicated.
But we can think about this in a follow up.

Reply:
Yeap I agree, we could change the format of the metadata that are returned to sth like this:
```
{
"cloud": {},
"orchestrator": {},
"other.ecs.fields": {},
}
```
Note: sth similar to what will do k8s metadata generators. All metadata are returned in top level as they are supposed to be attached in the event.


With this, the data manipulation at https://github.com/elastic/beats/blob/8a8c6d1bde023b4837a0b2d22f7e87a2ed202ea1/libbeat/processors/add_cloud_metadata/add_cloud_metadata.go#L115-L141 would be more smooth. I'm on favour of doing this in a followup PR since this one is already biiig. What do you think @jsoriano @kaiyan-sheng ?

Reply:
Yep, +1 to do this in a follow up.